packages/crypto/src/getAddresses.ts

```diff
@@ -0,0 +1,85 @@
+import { BtcAddress } from '@btckit/types';
+import { BitcoinAccount, ecdsaPublicKeyToSchnorr } from '@leather-wallet/bitcoin/src/bitcoin.utils';
+import { NetworkConfiguration } from '@leather-wallet/constants';
+import { bytesToHex } from '@stacks/common';
+import { TransactionPayload } from '@stacks/connect';
+
+import { useCurrentAccountNativeSegwitIndexZeroSigner } from './utils/bitcoin/native-segwit-account.hooks';
+import { useCurrentAccountTaprootIndexZeroSigner } from './utils/bitcoin/taproot-account.hooks';
+import { useCurrentStacksAccount } from './utils/stacks/stacks-account';
+import { StacksAccount } from './utils/stacks/stacks-account.models';
+import { KeyConfig } from './utils/useWalletType';
+
+export function useGetAddresses({
+  signatureIndex,
+  currentAccountIndex,
+  currentNetwork,
+  hasLedgerKeys,
+  hasSwitchedAccounts,
+  segwitAccount,
+  stacksAccounts,
+  taprootAccount,
+  transactionPayload,
+  wallet,
```

Comment:
This implementation means that the `@leather/crypto` package is "aware" of all the chains we support.

Say we had more chains in future, we'd have to come to this function an expand it with more object properties. What if this factory hook instead was responsible only for formatting, and delivering the response?

Then the API might look something like:

```ts
interface UseGetAddressesProps {
  onApproveGetAddresses(accounts: accounts[]): Promise<void>;
  accounts[]: SomeMoreGenericTypeForAnyChain;
}
function useGetAddresses({ onApproveGetAddresses, accounts }: UseGetAddressesProps) {}

// extension
useGetAddreseses({ 
  // type checking ensures these conform to standard
  accounts,
  onApproveGetAddresses(accounts) {
    // specific extension
    chrome.tabs.sendMessage(tabId, accounts);
  }
})

// RN
useGetAddreseses({ 
  accounts,
  onApproveGetAddresses(accounts) {
    // specific RN api
    somethingElseThatCommunitcatesWithWebView();
  }
})
```

Here, the responsibility is more scoped to just formatting, checking, and passing on the response.